contribute/design/proposal-volume-snapshots-clone.md

```diff
@@ -84,14 +101,20 @@ Along wih the OpenEBS Volume Dynamic Provisioner, OpenEBS Snapshot Provisioner w
 * OpenEBS Snapshot Provisioner will check the volume type:
   - If Jiva - delete the jivaService, jivaController and jivaReplica
   - If cStor:
-    * delete cStorService and cStoreController and associated configmap. 
-    * delete the cStorReplica (CRD)
-    * cstor-pool-mgmt side care will delete the cloned zvol. 
+    * delete cStorService and cStoreController and associated cStorVolume CR. 
+    * delete the cStorReplica CR
+    * cstor-pool-mgmt side car will delete the cloned zvol. 
 
 ### Revert the state of OpenEBS Volume to a previous snapshot
+This option is a distruptive operation.
+* Admin will have stopped the workload (if possible) that is using the volume.
 * Admin will use mayactl to list the snapshot on the given volume
 * Admin will select the snapshot to revert to. 
 * mayactl will do the following:
   - issue commands to delete the snapshot via kubectl, that were taken after the selected snapshot (updating the k8s db)
-  - issues REST API to either jivaController or cStorController(volume-mgmt sidecar?) to revert the state to selected snapshot
+  - update the cStorVolumeSnapshot CR to "revert". 
+    * cstor-volume-mgmt will stop the target and update the status to "stopped"
+    * cstor-pool-mgmt will rollback the zvol to the snapshot for the corresponding volumes. 
```

Comment:
thinking of - how to handle in case of any rollback failures in few nodes. Another point is - I guess - checks will be made to see that snapshots are available on all nodes.